contracts/RewardsBooster.sol

```diff
@@ -470,23 +470,80 @@ contract RewardsBooster is Initializable, OwnableUpgradeable, IRewardsBooster {
      * @notice Add Labor(online status) for current era project pool
      * @param _deploymentIds deployment id
      * @param _runners all runner addresses
-     * @param _missedLabors all missed labor seconds
+     * @param _disableds whether runner is disabled for reward
+     * @param _lastReportTimes last report time based on which missedLaborChanges were calculated
+     * @param _missedLaborChanges all missed labor within the current report period
      */
     function setMissedLabor(
         bytes32[] calldata _deploymentIds,
         address[] calldata _runners,
-        uint256[] calldata _missedLabors
+        bool[] calldata _disableds,
+        uint256[] calldata _lastReportTimes,
+        uint256[] calldata _missedLaborChanges,
+        uint256 _reportAt
     ) external {
         require(reporters[msg.sender], 'RB004');
+        require(
+            _deploymentIds.length == _runners.length &&
+                _deploymentIds.length == _disableds.length &&
+                _deploymentIds.length == _lastReportTimes.length &&
+                _deploymentIds.length == _missedLaborChanges.length,
+            'G020'
+        );
 
         for (uint256 i = 0; i < _runners.length; i++) {
-            DeploymentPool storage deployment = deploymentPools[_deploymentIds[i]];
-            RunnerDeploymentReward storage runnerDeplReward = deployment.runnerAllocationRewards[
-                _runners[i]
-            ];
-            runnerDeplReward.missedLaborTime = _missedLabors[i];
-            emit MissedLabor(_deploymentIds[i], _runners[i], _missedLabors[i]);
+            RunnerDeploymentReward storage runnerDeplReward = deploymentPools[_deploymentIds[i]]
+                .runnerAllocationRewards[_runners[i]];
+            require(_lastReportTimes[i] == runnerDeplReward.lastReportMissedLaborTime, 'RB008');
```

Comment:
Is it necessary to have `_lastReportTimes` in the function params, seems it only used for this check, the contract already have the `lastReportMissedLaborTime`, what's the purpose for this condition check?

Reply:
It is a guard so reporter won't mistakenly report a missedlabor twice.
Every time it is reported, lastReportMissedLaborTime will be refreshed.

Reply:
In that case if change `_reportAt >= runnerDeplReward.lastMissedLaborReportAt` to `_reportAt > runnerDeplReward.lastMissedLaborReportA` should achieve the same effect?

Reply:
I think you are right